pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1180,36 +1180,40 @@ func (c *PodAffinityChecker) InterPodAffinityMatches(pod *v1.Pod, meta algorithm
 	return true, nil, nil
 }
 
-// podMatchesPodAffinityTerms checks if the "targetPod" matches the given "terms"
-// of the "pod" on the given "nodeInfo".Node(). It returns three values: 1) whether
-// targetPod matches all the terms and their topologies, 2) whether targetPod
-// matches all the terms label selector and namespaces (AKA term properties),
-// 3) any error.
-func (c *PodAffinityChecker) podMatchesPodAffinityTerms(pod, targetPod *v1.Pod, nodeInfo *schedulercache.NodeInfo, terms []v1.PodAffinityTerm) (bool, bool, error) {
-	if len(terms) == 0 {
-		return false, false, fmt.Errorf("terms array is empty")
-	}
-	props, err := getAffinityTermProperties(pod, terms)
+// podMatchesPodAffinityTerms checks if the "targetPod" matches the given "affinityKind"
+// of the "pod" on the given "nodeInfo".Node(). It returns four values: 1) whether
+// targetPod matches any of the affnityTerms and their topologies, 2) whethertargetPod matches
+// all the affnityTerms and their topologies, 3) whether targetPodmatches all the terms label
+// selector and namespaces (AKA term properties), 4) any error.
+func (c *PodAffinityChecker) podMatchesPodAffinityTerms(pod, targetPod *v1.Pod, nodeInfo *schedulercache.NodeInfo, affinityKind string) (bool, bool, bool, error) {
```

Comment:
I'm not a fan of passing into a plain string to represent "kind". In this case, it's binary - affinity, or anti-affinity. So bool should be a better bet.

BTW: if we have more than 2 options, usually we create a custom type `type XYZ string`.